### PR TITLE
Don't store is value when it won't be useful

### LINKED ISSUE #659: Non-builtin custom elements which don't have definition yet, may get non-null is-value
https://github.com/whatwg/dom/pull/566

It is weird that the existence of CE definition during creation time affects whether is value is then later serialized.

### PR DESCRIPTION
**Pull request by domenic**
Fixes #659.

I believe the impact of this is entirely on the serialization of elements created this way.

EDIT: this also impacts whether `:defined` matches.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Jan 15, 2021, 7:33 AM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fwhatwg%2Fdom%2Fd7d69f121ba81d21b4d98a66e915387649132c51%2Fdom.bs&force=1&md-status=LS-PR&md-Text-Macro=PR-NUMBER%20661)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20whatwg/dom%23661.)._
</details>

### CODE REVIEW

**Comment by smaug----** [comment]
this is for https://github.com/whatwg/dom/issues/659, right?

**Comment by domenic** [comment]
Right, sorry, got them mixed up. Will edit OP/commit message.

File: dom.bs [hunk 1]
```diff
@@ -6052,6 +6052,15 @@ given a <var>document</var>, <var>localName</var>, <var>namespace</var>, and opt
    <li><p>Let <var>interface</var> be the <a>element interface</a> for <var>localName</var> and
    <var>namespace</var>.
 
+   <li>
+    <p>If <var>localName</var> is a <a>valid custom element name</a>, or <var>is</var> is not a
+    <a>valid custom element name</a>, then set <var>is</var> to null.</p>
+
+    <p class="note">In this case, even though <var>is</var> was supplied, there's no way that this
+    element will ever be <a lt="upgrade an element">upgraded</a>, so we shouldn't store the
```
Comment (annevk) [inline dom.bs]: there's -> there is

shouldn't -> will not

File: dom.bs [hunk 2]
```diff
@@ -6052,6 +6052,15 @@ given a <var>document</var>, <var>localName</var>, <var>namespace</var>, and opt
    <li><p>Let <var>interface</var> be the <a>element interface</a> for <var>localName</var> and
    <var>namespace</var>.
 
+   <li>
+    <p>If <var>localName</var> is a <a>valid custom element name</a>, or <var>is</var> is not a
+    <a>valid custom element name</a>, then set <var>is</var> to null.</p>
+
```
[the remark this answers is not in the document]
Comment (annevk) [reply dom.bs]: Any updates on that?

**Comment by annevk** [comment]
@smaug---- @domenic any updates here?

**Comment by domenic** [comment]
I'm happy to fix the grammatical issues here and file a bug on Chrome if @smaug---- can commit to updating the tests.

However, it appears that currently Chrome and Firefox agree on all cases in https://wpt.fyi/results/custom-elements/pseudo-class-defined.html?label=master&product=chrome%5Bexperimental%5D&product=edge&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&aligned, so I'm unsure whether we want to churn browsers in this way. Up to @smaug----.

**Comment by smaug----** [comment]
I can commit on updating tests.
Would the change require tiny tweak to elements created by the parser?

**Comment by annevk** [comment]
Looking at https://html.spec.whatwg.org/#create-an-element-for-the-token, I'm not sure. It won't be able to find a weird definition as that also uses the local name and it eventually delegates to this algorithm.

**Comment by annevk** [comment]
@domenic ping.

**Comment by domenic** [comment]
I've lost too much context to come back to this, I'm sorry to say. And it appears we never got the relevant tests written. I'll close this, and we can leave #659 open for someone else to fix someday.

**Pull request closed without merging**